pkg/subctl/cmd/show.go

```diff
@@ -17,35 +17,70 @@ limitations under the License.
 package cmd
 
 import (
+	"fmt"
 	"github.com/spf13/cobra"
-
+	submarinerclientset "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
 	"github.com/submariner-io/submariner-operator/pkg/discovery/network"
 )
 
-// infoCmd represents the info command
-var infoCmd = &cobra.Command{
-	Use:   "info",
+// showCmd represents the show command
+var showCmd = &cobra.Command{
+	Use:   "show",
 	Short: "Get information on your cluster related to submariner",
 	Long: `This command shows the status of submariner in your cluster,
 and the relevant network details from your cluster.`,
-	Run: clusterInfo,
+	Args: func(cmd *cobra.Command, args []string) error {
+		return checkShowArguments(args)
+	},
+	Run: runShowCmd,
 }
 
 func init() {
-	addKubeconfigFlag(infoCmd)
-	rootCmd.AddCommand(infoCmd)
+	addKubeconfigFlag(showCmd)
+	rootCmd.AddCommand(showCmd)
 }
 
-func clusterInfo(cmd *cobra.Command, args []string) {
+type ShowCmdArgType string
+
+const (
+	networks ShowCmdArgType = "networks"
+)
+
+func checkShowArguments(args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("'subctl show' argument must be specified")
+	}
 
+	showArg := ShowCmdArgType(args[0])
+	switch showArg {
+	case networks:
+		return nil
+	}
+	return fmt.Errorf("%v is not a valid 'subctl show' argument", args[0])
+}
+
+func runShowCmd(cmd *cobra.Command, args []string) {
+	showArg := ShowCmdArgType(args[0])
+	switch showArg {
+	case networks:
+		showNetwork()
+	}
+}
+
+func showNetwork() {
 	config, err := getRestConfig(kubeConfig, kubeContext)
+
 	exitOnError("Error connecting to the target cluster", err)
 	dynClient, clientSet, err := getClients(config)
 	exitOnError("Error connecting to the target cluster", err)
 
-	clusterNetwork, err := network.Discover(dynClient, clientSet)
+	submarinerClient, err := submarinerclientset.NewForConfig(config)
+	exitOnError("Unable to get the Submariner client", err)
+
+	clusterNetwork, err := network.Discover(dynClient, clientSet, submarinerClient)
 	exitOnError("There was an error discovering network details for this cluster", err)
 
 	clusterNetwork.Show()
-
+	exitOnError("There was an error discovering network details for this cluster", err)
```

Comment:
`err` hasn’t changed since the previous `exitOnError`, so this one isn’t necessary.